snuba/datasets/generic_metrics_processor.py

```diff
@@ -86,24 +98,40 @@ def process_message(
         timestamp = _ensure_valid_date(datetime.utcfromtimestamp(message["timestamp"]))
         assert timestamp is not None, "Invalid timestamp"
 
+        try:
+            retention_days = enforce_retention(message["retention_days"], timestamp)
+        except EventTooOld:
+            return None
+
         keys = []
-        indexed_values = []
+        indexed_values: MutableSequence[int] = []
+        raw_values: MutableSequence[str] = []
         tags = message["tags"]
         assert isinstance(tags, Mapping), "Invalid tags type"
 
         sorted_tag_items = sorted(tags.items())
         for key, value in sorted_tag_items:
-            assert key.isdigit() and isinstance(value, int), "Tag key/value invalid"
-            keys.append(int(key))
-            indexed_values.append(value)
+            assert key.isdigit() and (
+                isinstance(value, int) or isinstance(value, str)
+            ), "Tag key/value invalid"
 
-        try:
-            retention_days = enforce_retention(message["retention_days"], timestamp)
-        except EventTooOld:
-            return None
+            keys.append(int(key))
+            if isinstance(value, int):
+                indexed_values.append(value)
+                raw_values.append("")
+            elif isinstance(value, str):
+                indexed_values.append(0)
+                raw_values.append(value)
 
         raw_values_index = self._get_raw_values_index(message)
-        raw_values = [raw_values_index.get(str(v), "") for v in indexed_values]
+        processed_raw_values: MutableSequence[str] = []
+        for offset, indexed_value in enumerate(indexed_values):
+            if indexed_value != 0:
```

Comment:
I think it would be better to trust whether the raw value is populated or not as 0 is a valid value. 
Though empty string is valid as well. 

I think a more robust strategy would be to decide whether we are in indexed mode or not depending on the type of the tag values column and stick to that as both 0 and "" are valid values.

Even better, I would not even implicitly trust the type. 
Just add a `mode` flag in the message from the indexer that tells us whether we are in indexed mode or not. Then trust that. Messing up that way would be much harder.

Reply:
Using the `version` field now to determine whether to interpret tag values are strings or ints